dev-requirements.txt

```diff
@@ -30,5 +30,5 @@ wheel~=0.40
 
 # Adapter specific dependencies
 mock~=5.0
-sasl~=0.3.1
 thrift_sasl~=0.4.3
+pure-sasl
```

Comment:
This should also align with `requirements.txt`. There's a separate PR to remove all of this duplication that a community member submitted. Perhaps we merge that first and then update this branch.